tests/check/check_test.go

```diff
@@ -33,6 +37,130 @@ func TestCheckMySQL(t *testing.T) {
+	cfg.Log.Level = "none"
+	cfg.Datastore.Engine = "memory"
+
+	observerLogger, logs := observer.New(zap.DebugLevel)
+	serverCtx := &run.ServerContext{
+		Logger: &logger.ZapLogger{
+			Logger: zap.New(observerLogger),
+		},
+	}
+
+	cancel := tests.StartServerWithContext(t, cfg, serverCtx)
+	defer cancel()
+
+	conn, err := grpc.Dial(cfg.GRPC.Addr,
+		grpc.WithBlock(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	client := openfgav1.NewOpenFGAServiceClient(conn)
+
+	createStoreResp, err := client.CreateStore(context.Background(), &openfgav1.CreateStoreRequest{
+		Name: "openfga-demo",
+	})
+	require.NoError(t, err)
+
+	storeID := createStoreResp.GetId()
+
+	typedefs := parser.MustParse(`
+	type user
+
+	type document
+	  relations
+	    define viewer: [user] as self
+	`)
+
+	writeModelResp, err := client.WriteAuthorizationModel(context.Background(), &openfgav1.WriteAuthorizationModelRequest{
+		StoreId:         storeID,
+		SchemaVersion:   typesystem.SchemaVersion1_1,
+		TypeDefinitions: typedefs,
+	})
+	require.NoError(t, err)
+
+	authorizationModelID := writeModelResp.GetAuthorizationModelId()
+
+	_, err = client.Write(context.Background(), &openfgav1.WriteRequest{
+		StoreId: storeID,
+		Writes: &openfgav1.TupleKeys{
+			TupleKeys: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:anne"),
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	type test struct {
+		_name           string
+		req             *openfgav1.CheckRequest
+		expectedContext map[string]interface{}
+	}
+
+	tests := []test{
+		{
+			_name: "check_success",
+			req: &openfgav1.CheckRequest{
+				AuthorizationModelId: authorizationModelID,
+				StoreId:              storeID,
+				TupleKey:             tuple.NewTupleKey("document:1", "viewer", "user:anne"),
+			},
+			expectedContext: map[string]interface{}{
+				"grpc_service":           "openfga.v1.OpenFGAService",
+				"grpc_method":            "Check",
+				"grpc_type":              "unary",
+				"grpc_code":              int32(0),
+				"raw_request":            fmt.Sprintf(`{"store_id":"%s","tuple_key":{"object":"document:1","relation":"viewer","user":"user:anne"},"contextual_tuples":null,"authorization_model_id":"%s","trace":false}`, storeID, authorizationModelID),
+				"raw_response":           `{"allowed":true,"resolution":""}`,
+				"authorization_model_id": authorizationModelID,
+				"store_id":               storeID,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test._name, func(t *testing.T) {
+			_, _ = client.Check(context.Background(), test.req)
+
+			filteredLogs := logs.Filter(func(e observer.LoggedEntry) bool {
+				if e.Message == "grpc_req_complete" {
+					for _, ctxField := range e.Context {
+						if ctxField.Equals(zap.String("grpc_method", "Check")) {
+							return true
+						}
+					}
+				}
+
+				return false
+			})
+
+			expectedLogs := filteredLogs.All()
+			require.Len(t, expectedLogs, 1)
```

Comment:
this assertion will break when we add more tests 😅

Reply:
fixed by clearing observed logs after reach run.